backend/src/index.ts

```diff
@@ -1 +1,60 @@
-console.log("hello world");
+import type { RoomData, BoardState } from "./types";
+
+const http = require("http");
+
+const express = require("express");
+const { Server } = require("socket.io");
+
+const app = express();
+const server = http.createServer(app);
+const io = new Server(server);
+const PORT = process.env.PORT || 3000;
+
+// boards maps room to BoardState (BoardState contains all data in regards to the state of the game and is declared in types.d.ts)
+const boards = new Map<string, BoardState>();
+
+// rooms maps username to RoomData (RoomData contains connection information for users and is declared in types.d.ts)
+const rooms = new Map<string, RoomData>();
+
+server.listen(PORT, () => {
+  console.log(`Server listening on ${PORT}`);
+});
+
+// triggered whenever a new socket connects to server
+// the new socket should send the username of the client through the auth object
+io.on("connection", (socket: any) => {
+  const username = socket.handshake.auth.username;
+  let currRoom = "Room " + boards.size;
+
+  // Check if user is already in a room
```

Comment:
Could you explain the purpose of detecting the presence of the user and what it does?